ios_ui/MovieRec/RecommendationsTableViewController.swift

```diff
@@ -66,23 +70,62 @@ class RecommendationsTableViewController: UITableViewController {
         makeAPICall()
     }
     
-    private func refreshMovies(data: Data) -> Void {
-        let responseJSON = try? JSONSerialization.jsonObject(with: data, options: [])
-        self.recommendations.removeAll()
-        if let responseJSON = responseJSON as? [String: [String]] {
-            for recommendation in responseJSON["recommendations"]! {
-                self.recommendations += [Recommendation(title: recommendation)]
-            }
+    private func refreshMovies(data: JSON) -> Void {
+        for recommendation in data.arrayValue {
+            self.recommendations += [Recommendation(title: recommendation.stringValue)]
+        }
 
-            DispatchQueue.main.async {
-                self.tableView.reloadData()
-            }
+        DispatchQueue.main.async {
+            self.tableView.reloadData()
         }
+        
         self.ratings.removeAll()
         delegate?.clearRatings()
         print(ratings.count)
     }
     
+    private func startJobPolling(data: Data) -> Void {
+        let responseJSON = try? JSONSerialization.jsonObject(with: data, options: [])
+        if let responseJSON = responseJSON as? [String: String] {
+            print(responseJSON["job_id"]!)
+            startTimer(job_id: responseJSON["job_id"]!)
+        }
+    }
+    
+    private func checkJobStatus(data: Data) -> Void {
+        let json = JSON(data: data)
+        if json["status"].stringValue == "completed" {
```

Comment:
I would actually create an enum of some sort that indicates the status and use its rawValue as its type to be more type safe instead of using a raw string.